src/webots/gui/WbConsole.cpp

```diff
@@ -397,74 +407,106 @@ void WbConsole::handlePossibleAnsiEscapeSequences(const QString &msg, WbLog::Lev
+      html = htmlSpan(msg.mid(0, i), level);
     while (1) {
       i += 2;  // skip the "\033[" chars
       QString sequence;
-      while (msg[i] != 'm')
-        sequence += msg[i++];
-      i++;
-
+      while (msg[i] != '\0' && msg[i] != 'm') {  // 'm' is the final char when supporting only colors
+        sequence += msg[i];
+        if (msg[i] == 'J' && msg[i - 1] == '2')  // "2J" is not terminated by 'm', but is still valid
+          break;
+        i++;
+      }
+      i++;                                           // moving to the next char for next iteration
       const QStringList codes(sequence.split(";"));  // handle multiple (e.g. sequence "ESC[0;39m" )
       foreach (const QString code, codes) {
-        // the stored sequence may be "0" or "1", "30", "31", "32", etc.
-        if (code == "0") {  // reset to default
-          mColor = ansiBlack();
-          mBold = false;
-        } else if (code == "1")  // bold
+        // the stored sequence may be "0" or "1", "4", "2J", "30", "31", "32", etc.
+        if (code == "0")  // reset to default
+          resetFormat();
+        else if (code == "1")  // bold
           mBold = true;
-        else if (code.startsWith("3")) {  // foreground color change
+        else if (code == "4")  // underlined
+          mUnderline = true;
+        else if (code.startsWith("2")) {  // clear console screen
+          const char c = code.toLocal8Bit().data()[1];
+          if (c == 'J') {             // code == "2J"
+            WbConsole::clear(false);  // perform a clear by preserving format
+            html.clear();             // nothing to output since clear has been done
+          }
+        } else if (code.startsWith("3")) {  // foreground color change
           const char c = code.toLocal8Bit().data()[1];
           switch (c) {
-            case '1':
-              mColor = ansiRed();
+            case '1':  // 31
+              mForegroundColor = ansiRed();
               break;
-            case '2':
-              mColor = ansiGreen();
+            case '2':  // 32
+              mForegroundColor = ansiGreen();
               break;
-            case '3':
-              mColor = ansiYellow();
+            case '3':  // etc...
+              mForegroundColor = ansiYellow();
               break;
             case '4':
-              mColor = ansiBlue();
+              mForegroundColor = ansiBlue();
               break;
             case '5':
-              mColor = ansiMagenta();
+              mForegroundColor = ansiMagenta();
               break;
             case '6':
-              mColor = ansiCyan();
+              mForegroundColor = ansiCyan();
               break;
             case '7':
-              mColor = ansiWhite();
+              mForegroundColor = ansiWhite();
               break;
             case '9':  // 39 - Default text color
             default:
-              mColor = ansiBlack();
+              mForegroundColor = ansiBlack();
```

Comment:
The default case should never happen hence we should do nothing here.

Reply:
True, I updated it to some more understandable version, thanks